### ISSUE TITLE
ocean endpoint as user config

### ISSUE THREAD

**Issue by adrian-schnell** [DeFiCh/wallet#2753]
#### What would you like to be added:
To give users the possibility to setup their own ocean and use this one instead of the official one, I'd love to see a config possibility to override the ocean base url. Maybe incl function test and automatic fallback to the official ocean.

#### Why is this needed:
As we're growing fast, more and more bots are used, the Ocean is hard at it's limit. To reduce the load on ocean and having a better user experice, this feature would be very handy.

**Comment by thedoublejay** [comment]
Hi @adrian-schnell, thanks for raising this. I believe it's a good idea and it can open up to better decentralization. However, there are some pitfalls also on having this option.

Currently, it's also possible by forking and just updating the code. Basically, just having another wallet to use a different interface.

One pitfall would be a security issue. Someone can just message a user on telegram, ask them to replace the link with a malicious API to steal their funds. So it's some of the things we need to consider for this issue.

**Comment by 0ptim** [comment]
Forking, building and always keeping up to date on updates is a very high effort I think.

Make the functionality somewhat hidden/hard to use. This will prevent normies from making any mistakes.

Furthermore, a clear warning would need to be displayed, where the custom endpoint could be configured. This way, users can quickly identify any scam attempts.

**Comment by thedoublejay** [comment]
@0ptim Yes, it will be designed like this. The goal would be to make it available but not as easy to do. So it will be only for "power users" in a sense. Thanks for the feedback.

**Comment by adrian-schnell** [comment]
> @0ptim Yes, it will be designed like this. The goal would be to make it available but not as easy to do. So it will be only for "power users" in a sense. Thanks for the feedback.

that's a great idea!
e.g. it's hidden in the "about settings" like the beta features before..

**Comment by thedoublejay** [comment]
> one more point.. maybe some user want to secure their own ocean with htaccess. would it be possible to enter username/password optionally in the wallet?

This requirement might be too complicated and will be used by a small number of users. I think the update of ocean API config should be a good start first.

**Comment by sancag** [comment]
Excellent request. Next CFP round will come with a host-yourself ocean solution. Currently I am re-writing my local DNS server to point to my own ocean. Having the capability to specify a custom URL would be fantastic.

I would enforce TLS i.e only allow urls with https

**Comment by pierregee** [comment]
@izzycsy @dwivedinalin 
If there are problems in connecting to our API (default endpoint), we display this message which points to our [API status site](https://status.defichain.com/)

![image](https://user-images.githubusercontent.com/13292662/174532547-846df27c-8ac4-4a73-804b-6dad3415f2e0.png)



For custom endpoints, this will be replaced with this warning if there's a problem with the connection 
**We have detected issues with your custom endpoint that is affecting your connection. You are advised to check on the status of your custom endpoint provider.**

**Comment by Graciaho** [comment]
Thank you for the submission. Reconfiguring ocean endpoint is now available on the latest release of Light Wallet v.1.14.0 and above.
Please note that users will need to opt into Beta Flag to enable this feature. To access Beta Flag screen, select  `Settings` > `About`

How to change from default endpoint to custom endpoint:
1. Under `Settings`, select `Server`. User should see the default endpoint URL selected
2. To change endpoint, select `Unlock` , and enter a valid custom endpoint URL
3. Enter 6-digit passcode when prompted for verification
4. Restart the Light Wallet app

How to reset to default endpoint: 
1. Under `Settings,` select `Server`
2. Select `Unlock`
3. Select `Reset`
4. A prompt will be sent to user to confirm the request to reset to default endpoint.. Select `Reset` to continue
5. Enter 6-digit passcode when prompted for verification
6. Restart the Light Wallet app
